crash/src/app/accident-report-container/map/map.component.ts

```diff
@@ -0,0 +1,188 @@
+    // this.loadAPI();
+    this.loadAPIMapscript()
+  }
+ 
+  options: google.maps.MapOptions = {
+    mapTypeId: 'hybrid',
+    zoomControl: false,
+    scrollwheel: false,
+    disableDoubleClickZoom: true,
+    maxZoom: 15,
+    minZoom: 8,
+  };
+
+  markerPositions: google.maps.LatLngLiteral[] = [{ lat: 53.540235028, lng: -113.49818175 }];// default to downtown edmonton
+  zoom = 11;
+  display: any;
+  center: google.maps.LatLngLiteral = {
+    lat: 53.540235028,
+    lng: -113.49818175
+  };
+  icon!: google.maps.Icon;
+  panelOpenState = false;
+  public map!: GoogleMap;
+  WeatherData!: any;
+  AddressData!: any;
+  public weathericon: string = '';
+  public location: string = '';
+  public weatherconditions: string = '';
+
+  @Output() m_weather = new EventEmitter<string>()
+  @Output() m_address = new EventEmitter<string>();
+  @Output() m_weathericon = new EventEmitter<string>();
+
+
+  moveMap(event: google.maps.MapMouseEvent) {
+    if (event.latLng != null) this.display = (event.latLng.toJSON());
+    if (event.latLng != null)
+      this.center.lat = event.latLng.lat();
+
+  }
+
+  zoomIn() {
+    if (this.options.maxZoom != null)
+      if (this.zoom < this.options.maxZoom) this.zoom++;
+  }
+
+  zoomOut() {
+    if (this.options.minZoom != null)
+      if (this.zoom > this.options.minZoom) this.zoom--;
+  }
+  click(event: google.maps.MapMouseEvent) {
+    if (event.latLng != null) this.display = event.latLng.toJSON();
+    if (event.latLng != null)
+      this.center.lng = event.latLng.lng();
+
+    this.markerPositions = [{ lat: this.center.lat, lng: this.center.lng }];
+  }
+
+  public getIcon(): google.maps.Icon {
+    let ret: google.maps.Icon = {
+      url: ''
+    }
+    return ret;
+  }
+  collectInformation() {
+
+    this.getWeather();
+    this.getInfoFromAddress(this.center);
+    //this.searchNearLandmark(this.center); -- placesAPI is call spits cross-origin error
+  }
+
+  getInfoFromAddress(latlng: any) {
+
+    let geocoder = new google.maps.Geocoder();
+    let infowindow = new google.maps.InfoWindow();
+    geocoder.geocode({ location: latlng })
+      .then((response) => {
+        if (response.results[0]) {
+
+          this.location = response.results[0].formatted_address;
+          infowindow.setContent(response.results[0].formatted_address);
+          this.m_address.emit(this.location);
+
+        } else {
+          window.alert("No results found");
+        }
+      })
+      .catch((e) => window.alert("Geocoder failed due to: " + e));
+
+
+  }
+  //RoadAPI calls requires premium subscription
+  getSpeedLimit(addresses: any) {
+    let places = "placeID=";
+    for (let i = 0; i < addresses.length; i++) {
+      places += addresses[i].place_id + "&"
+    }
+
+  }
+  searchNearLandmark(latlng: any) {
+    this.crashservice.getNearbyPlaces(latlng.lat, latlng.lng)
+      .subscribe(
+        (res: any) => {
+          console.log(res.results);
```

Comment:
I think we can keep this one for now, but please try to avoid using console logs